doc/fundamentals/distributed_computing.md

```diff
@@ -0,0 +1,115 @@
+
+Instead of confining users to data sharding patterns such as Map (split everything as much as possible) and Reduce (all the data must be grouped together), Pachyderm uses [Glob Patterns](https://en.wikipedia.org/wiki/Glob_(programming)) to offer incredible flexibility in defining your data distribution. 
+
+Glob patterns are defined for each input of a pipeline. 
+
+```
+"inputs": [
+    {
+      "repo": {
+        "name": "string"
+      },
+      "glob": "string"
+    }
+  ]
+```
+That means you could easily define two inputs, one with the data highly distributed and another where it's grouped together. 
+
+### Datums
+
+Pachyderm uses the glob pattern to determine how many “datums” an input consists of. Datums are the unit of parallelism in Pachyderm. That is, Pachyderm attempts to process datums in parallel whenever possible.
+
+If you have two workers and define 2 datums, Pachyderm will send one datum to each worker. In a scenario where there are more datums than workers, Pachyderm will queue up extra datums and send them to workers as they finish processing previous datums. 
+
+### Glob Patterns
+
+Intuitively, you should think of the input repo as a file system where the glob pattern is being applied to the root of the file system. The files and directories that match the glob pattern are considered datums.
+
+For example, a glob pattern of just `/` would denote the entire input repo as a single datum. All of the input data would be given to a single worker similar to a typical reduce-style operation.
+
+Another commonly used glob pattern is `/*`. `/*` would define each top level object (file or directory) in the input repo as its own datum. If you have a repo with just 10 files in it and no directory structure, every file would be a datum and could be processed independently. This is similar to a  typical map-style operation.
+
+But Pachyderm can do anything in between too. If you had a directory structure with each state as a directory and a file for each city such as:
+
+```
+/California
+   /San-Francisco.json
+   /Los-Angeles.json
+   ...
+/Colorado
+   /Denver.json
+   /Boulder.json
+   ...
+...
+```
+And you need to process all the data for a given state together, `/*` would also be the correct glob pattern. You'd have one datum per state, meaning all the cities for a given state would be processed together by a single worker, but each state can be processed independently. 
+
+If we instead used the glob pattern `/*/*` for the states example above, each `city.json` would be it's own datum. 
+
+Glob patterns also let you take only a particular directory (or subset of directories) as inputs instead of the whole input repo. 
+
+If we create a pipeline that is specifically only for California, we can use a glob pattern of `/California/*` to only use the data in that directory as input to our pipeline. 
+
+## Only Processing New Data
```

Comment:
I'm on the fence, but I'm fine with that change. The only reason I can think of for keeping it as `##` is for discoverability because this is asked about a ton.